src/expr/src/scalar/like_pattern.rs

```diff
@@ -272,6 +277,41 @@ fn build_subpatterns(pattern: &str, escape: char) -> Result<Vec<Subpattern>, Eva
     Ok(subpatterns)
 }
 
+/// Builds a regular expression that matches some parsed Subpatterns.
+fn build_regex(subpatterns: &[Subpattern], case_insensitive: bool) -> Result<Regex, EvalError> {
+    let mut r = String::from("^");
+    for sp in subpatterns {
+        if sp.consume == 0 && sp.many {
+            r.push_str(".*");
+        } else if sp.consume == 1 {
+            r.push('.');
+            if sp.many {
+                r.push('+');
+            }
+        } else if sp.consume > 1 {
+            r.push_str(".{");
+            write!(&mut r, "{}", sp.consume).unwrap();
+            if sp.many {
+                r.push(',');
+            }
+            r.push('}');
+        }
+        regex_syntax::escape_into(&sp.suffix, &mut r);
+    }
```

Comment:
It doesn't much matter, but it's a smidge harder to follow this logic vs. the original logic IMO! Was there a particular benefit to rewriting this to build the regex off of the subpatterns, rather than just building it off the whole pattern string?

Reply:
👍🏽 sgtm. It's like maybe 10% harder to follow (I had to go look up what `{1,}` means), which seems totally worth it if you've got future optimizations planned on the new structure. Just wanted to understand the thinking!